### DISCUSSION TITLE
Record calls to a third party contact center

### DISCUSSION THREAD

**Discussion by chrisansotti** [amazon-connect/amazon-connect-salesforce-scv#132] [Q&A] [answered]
We utilize an overflow contact center that receives calls when our in-house queues are above a certain threshold. Currently in RingCentral we simply have an extension that is set up with the phone number for the overflow center and since the extension resides within the call center it continues to record the call even after the agent disconnects.

Amazon Connect obviously doesn't utilize extensions in this way and their support tells me that the recording stops as soon as the agent disconnects, but I'm curious if anyone has found a workaround to this.

**Comment by dougjaso** [comment, marked as answer]
This isn't specifically a Service Cloud Voice issue, more of a general Connect one. Connect was designed to only record conversations connected to Agents. If you want to record IVR, transfers, or anything else, you need to initialize streams and process that audio segment post call. For example, 

1. Quick Connect as a transfer to queue. Queue is irrelevant, contact flow is important. 
2. In contact flow, start media streaming, then transfer to phone number
3. Upon call completion, process CTR for the transfer leg, and extract the audio from the KVS stream using the data in the CTR to get the stream and fragment info. 

Sample external transfer CTR with the data:
```
{
   "AWSAccountId":"REDACTED",
   "AWSContactTraceRecordFormatVersion":"2017-03-10",
   "Agent":"None",
   "AgentConnectionAttempts":0,
   "AnsweringMachineDetectionStatus":"None",
   "Attributes":{
      "external-streaming":"1"
   },
   "Campaign":{
      "CampaignId":"None"
   },
   "Channel":"VOICE",
   "ConnectedToSystemTimestamp":"2022-08-23T20:25:33Z",
   "ContactDetails":{
      
   },
   "ContactId":"REDACTED-5551c53b749a",
   "CustomerEndpoint":{
      "Address":"+1REDACTED",
      "Type":"TELEPHONE_NUMBER"
   },
   "DisconnectReason":"THIRD_PARTY_DISCONNECT",
   "DisconnectTimestamp":"2022-08-23T20:25:54Z",
   "InitialContactId":"REDACTED-afe2e401b2b4",
   "InitiationMethod":"TRANSFER",
   "InitiationTimestamp":"2022-08-23T20:25:33Z",
   "InstanceARN":"arn:aws:connect:us-west-2:7REDACTED:instance/REDACTED-3f088df00647",
   "LastUpdateTimestamp":"2022-08-23T20:27:00Z",
   "MediaStreams":[
      {
         "Type":"AUDIO"
      }
   ],
   "NextContactId":"None",
   "PreviousContactId":"REDACTED-afe2e401b2b4",
   "Queue":"None",
   "Recording":"None",
   "Recordings":[
      {
         "DeletionReason":"None",
         "FragmentStartNumber":"91343852333181462103243005496983963028219882580",
         "FragmentStopNumber":"91343852333181462182471168011248304744694049367",
         "Location":"arn:aws:kinesisvideo:us-west-2:REDACTED:stream/REDACTED-b7545daae62f/1625074127227",
         "MediaStreamType":"VIDEO",
         "ParticipantType":"CUSTOMER",
         "StartTimestamp":"2022-08-23T20:25:37Z",
         "Status":"None",
         "StopTimestamp":"2022-08-23T20:25:54Z",
         "StorageType":"KINESIS_VIDEO_STREAM"
      }
   ],
   "References":[
      
   ],
   "ScheduledTimestamp":"None",
   "SystemEndpoint":{
      "Address":"+1REDACTED",
      "Type":"TELEPHONE_NUMBER"
   },
   "TransferCompletedTimestamp":"2022-08-23T20:25:44Z",
   "TransferredToEndpoint":{
      "Address":"+1REDACTED",
      "Type":"TELEPHONE_NUMBER"
   },
   "VoiceIdResult":"None"
}
```
[This page](https://docs.aws.amazon.com/connect/latest/adminguide/access-media-stream-data.html) can help with how to access the audio data in KVS.
